lib/src/robot/client.dart

```diff
@@ -1,36 +1,35 @@
 import 'package:grpc/grpc_connection_interface.dart';
-import 'package:viam_sdk/src/gen/robot/v1/robot.pbgrpc.dart';
-import 'package:viam_sdk/src/resource/manager.dart';
-import 'package:viam_sdk/viam_sdk.dart';
 
+import '../domain/web_rtc/web_rtc_client/web_rtc_client.dart';
+import '../gen/common/v1/common.pb.dart';
+import '../gen/robot/v1/robot.pbgrpc.dart';
 import '../media/stream/client.dart';
+import '../resource/base.dart';
+import '../resource/manager.dart';
+import '../resource/registry.dart';
+import '../rpc/dial.dart';
+import '../viam_sdk.dart';
 
 class RobotClientOptions {
-  bool disableWebRtc = false;
-  String locationSecret;
-  bool insecure = false;
+  DialOptions dialOptions;
 
-  RobotClientOptions(this.disableWebRtc, this.insecure, this.locationSecret);
-
-  RobotClientOptions.withSecret(this.locationSecret);
+  RobotClientOptions.withLocationSecret(String locationSecret)
+      : dialOptions = DialOptions()..credentials = Credentials.locationSecret(locationSecret);
 }
 
 class RobotClient {
   late ClientChannelBase channel;
   late RobotServiceClient _client;
   List<ResourceName> resourceNames = [];
   ResourceManager _manager = ResourceManager();
-  Map<String, StreamClient> _streams = {};
-  late Viam viam;
+  final Map<String, StreamClient> _streams = {};
+  Viam? viam;
```

Comment:
[minor] I think this is unused, remove?

a viam is referenced on lines 38-41, but that name is overloaded and the new one will be used and not this one.